### PR TITLE
Add optional HttpResponse parameter to ReplyForbiddenWithWwwAuthenticateHeaderAsync. Handle null context.

### PR DESCRIPTION
**Pull request by pmaytak**
Issue #380

### CODE REVIEW

File: src/Microsoft.Identity.Web/ITokenAcquisition.cs [hunk 1]
```diff
@@ -51,9 +51,11 @@ public interface ITokenAcquisition
         /// </summary>
         /// <param name="scopes">Scopes to consent to.</param>
         /// <param name="msalSeviceException"><see cref="MsalUiRequiredException"/> triggering the challenge.</param>
+        /// <param name="httpResponse">The <see cref="HttpResponse"/> to update.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
         Task ReplyForbiddenWithWwwAuthenticateHeaderAsync(
             IEnumerable<string> scopes,
-            MsalUiRequiredException msalSeviceException);
+            MsalUiRequiredException msalSeviceException,
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/ITokenAcquisition.cs:58]: there is a typo (pre-existing). msalSeviceException => msalServiceException.

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 2]
```diff
@@ -533,8 +533,9 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
         /// </summary>
         /// <param name="scopes">Scopes to consent to.</param>
         /// <param name="msalServiceException">The <see cref="MsalUiRequiredException"/> that triggered the challenge.</param>
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs:535]: same thing here.

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 3]
```diff
@@ -562,9 +563,10 @@ public async Task ReplyForbiddenWithWwwAuthenticateHeaderAsync(IEnumerable<strin
 
             string parameterString = string.Join(", ", parameters.Select(p => $"{p.Key}=\"{p.Value}\""));
 
-            if (CurrentHttpContext != null)
+            httpResponse ??= CurrentHttpContext?.Response;
+
+            if (httpResponse != null)
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/TokenAcquisition.cs:568]: if the httpResponse is null, do we want to throw with an aka.ms link?

File: src/Microsoft.Identity.Web/ITokenAcquisition.cs (hunk 1 above)
Comment (pmaytak) [reply src/Microsoft.Identity.Web/ITokenAcquisition.cs:58]: @jmprieur Do we want to include this breaking change into 0.2.2 or into 0.3.0?

Comment (jmprieur) [reply src/Microsoft.Identity.Web/ITokenAcquisition.cs:58]: you mean the name of the parameter?
I would think yes (we are in preview and I doubt many people would have used the named: way of calling the API.
I can take the risk @pmaytak

File: src/Microsoft.Identity.Web/TokenAcquisition.cs (hunk 3 above)
Comment (pmaytak) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:568]: @jmprieur That was my initial proposal - throw an Exception, something like InvalidOperationException. I think just a descriptive message will suffice (ex. HttpContext is null, pass HttpResponse as a parameter). What's your idea of what to put into a wiki article about this?

Comment (jmprieur) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:568]: @pmaytak : Let's throw an InvalidOperationException with the message your proposed. Good idea!

**Pull request merged**